app-text/cherrytree/cherrytree-0.39.2.ebuild

```diff
@@ -5,7 +5,7 @@ EAPI=7
 
 PYTHON_COMPAT=( python3_{6,7,8} )
 
-inherit autotools desktop python-any-r1 xdg-utils
+inherit autotools desktop python-r1 xdg-utils
```

Comment:
I just download the sources for 0.39.2 and it seems to be using setup.py, why wouldn't this work with *distutils-r1.eclass*?

Reply:
Oh...

It's an alpha C++ version which located in a [separate 'future' directory](https://github.com/giuspen/cherrytree/tree/master/future) inside of the master branch

I need to mask/drop old 0.39.* versions because these packages require deprecated `dev-python/pygtksourceview` and, after that, I need to ~bump~ it to untagged [0.1.0 version](https://github.com/giuspen/cherrytree/blob/c1b48d563e2c44b9df385f92f461bed58a4e2d26/future/CMakeLists.txt#L4)

More: 
https://bugs.gentoo.org/708094
https://bugs.gentoo.org/706462
https://bugs.gentoo.org/694800

Reply:
I have a prototype of a future version of ebuild (cherrytree-0.1.0_p20200505.ebuild) which I want to add
```bash
# Copyright 1999-2020 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

EAPI=7

PYTHON_COMPAT=( python3_{6,7,8} )

HASH_COMMIT="26afcc6b27635d5b710c481daf9ae74c8ab20ea8" # 20200505

inherit cmake desktop python-r1 xdg-utils

DESCRIPTION="A hierarchical note taking application (C++ version)"
HOMEPAGE="https://www.giuspen.com/cherrytree"
SRC_URI="https://github.com/giuspen/cherrytree/archive/${HASH_COMMIT}.tar.gz -> ${P}.tar.gz"

LICENSE="GPL-3"
SLOT="0"
KEYWORDS="~amd64 ~x86"
IUSE="nls test"
REQUIRED_USE="${PYTHON_REQUIRED_USE}"

RESTRICT="!test? ( test )"

RDEPEND="${PYTHON_DEPS}
	app-text/gspell
	dev-cpp/gtkmm:3.0
	dev-cpp/gtksourceviewmm:3.0
	dev-cpp/libxmlpp:2.6
	dev-cpp/pangomm
	dev-db/sqlite:3
	dev-libs/glib:2
	dev-libs/libxml2:2"
DEPEND="${RDEPEND}
	dev-python/lxml[${PYTHON_USEDEP}]"
BDEPEND="${PYTHON_DEPS}
	dev-util/intltool
	virtual/pkgconfig
	nls? ( sys-devel/gettext )
	test? ( dev-util/cpputest )"

S="${WORKDIR}/${PN}-${HASH_COMMIT}/future"

pkg_setup() {
	# setup py3 executables for future/scripts/*.py
	python_setup
}

src_prepare() {
	python_fix_shebang "${S}/scripts"

	sed -i \
		-e "s|\(CT_VERSION.*{\)\(.*\)\(};\)$|\1\"${PV}\"\3|" \
		src/ct/ct_const.h || die

	# TODO: It's an alpha/future C++ version and tests is enabled by default
	# We need to check any changes of it in CMakeLists and remove this lines
	# before bump a new version
	#
	# Disable cpptest pkg checking if 'test' flag is inactive
	if ! use test; then
		sed -i \
			-e '/^pkg_check_modules(CPPUTEST cpputest  REQUIRED)/d' \
			-e '/^enable_testing()/d;/^add_subdirectory(tests)/d' \
			CMakeLists.txt || die
	fi

	cmake_src_prepare
}

pkg_postinst() {
	xdg_icon_cache_update
	xdg_mimeinfo_database_update
	xdg_desktop_database_update
}

pkg_postrm() {
	xdg_icon_cache_update
	xdg_mimeinfo_database_update
	xdg_desktop_database_update
}
```